src/arithmetic/rewrite_simplify.cc

```diff
@@ -626,6 +671,590 @@ Mutate_(const Mod* op, const Expr& self) {
+    TVM_TRY_REWRITE(x - y < x - z, z < y);
+
+    TVM_TRY_REWRITE(x < x + z, 0 < z);
+    TVM_TRY_REWRITE(x < z + x, 0 < z);
+    TVM_TRY_REWRITE(x < x - z, z < 0);
+    TVM_TRY_REWRITE(c1 < x + c2, c1 - c2 < x);
+    TVM_TRY_REWRITE(c1 < c2 - x, x < c2 - c1);
+
+    TVM_TRY_REWRITE_IF(x * c1 < y * c1, x < y,
+                       c1.Eval()->value > 0);
+
+    // require c1 > 0 to work for any div mode
+    TVM_TRY_REWRITE_IF(x * c2 < c1, x < (c1 - 1) / c2 + 1,
+                       c1.Eval()->value > 0 &&
+                       c2.Eval()->value > 0);
+    TVM_TRY_REWRITE_IF(x / c1 < c2, x < c1 * c2,
+                       c1.Eval()->value > 0 &&
+                       c2.Eval()->value > 0);
+
+    TVM_TRY_REWRITE_IF(c1 < x * c2, c1 / c2 < x,
+                       c1.Eval()->value >= 0 &&
+                       c2.Eval()->value > 0);
+    TVM_TRY_REWRITE_IF(c1 < x / c2, (c1 + 1) * c2 - 1 < x,
+                       c1.Eval()->value >= 0 &&
+                       c2.Eval()->value > 0);
+
+    // division related simplificationx
+    // invariance for any div mod: x - (x / c1) * c1 == x % c1
+    TVM_TRY_REWRITE_IF((x / c1) * c1 < x, 0 < x % c1,
+                       c1.Eval()->value > 0);
+    TVM_TRY_REWRITE_IF((x / c1) * c1 < x + y, 0 < x % c1 + y,
+                       c1.Eval()->value > 0);
+    TVM_TRY_REWRITE_IF((x / c1) * c1 < x - y, y < x % c1,
+                       c1.Eval()->value > 0);
+
+    TVM_TRY_REWRITE_IF(((x + c2)/ c1) * c1 < x,
+                       c2 < (x + c2) % c1,
+                       c1.Eval()->value > 0);
+    TVM_TRY_REWRITE_IF(((x + c2)/ c1) * c1 < x + y,
+                       c2 < (x + c2) % c1 + y,
+                       c1.Eval()->value > 0);
+    TVM_TRY_REWRITE_IF(((x + c2)/ c1) * c1 < x - y,
+                       y < (x + c2) % c1 + (0 - c2),
+                       c1.Eval()->value > 0);
+
+    // canonicalization rule
+    TVM_TRY_RECURSIVE_REWRITE(min(x, y) < z, x < z || y < z);
+    TVM_TRY_RECURSIVE_REWRITE(max(x, y) < z, x < z && y < z);
+    TVM_TRY_RECURSIVE_REWRITE(z < min(x, y), z < x && z < y);
+    TVM_TRY_RECURSIVE_REWRITE(z < max(x, y), z < x || z < y);
+
+    TVM_TRY_REWRITE(x - c1 < 0, x < c1);
+    TVM_TRY_REWRITE(x + c1 < c2, x < c2 - c1);
+  }
+  return ret;
+}
+
+Expr RewriteSimplifier::Impl::
+Mutate_(const Not* op, const Expr& self) {
+  Expr ret = IRMutator::Mutate_(op, self);
+  op = ret.as<Not>();
+  Expr const_res = TryConstFold<Not>(op->a);
+  if (const_res.defined()) return const_res;
+  // Pattern var to match any expression
+  PVar<Expr> x, y;
+  PVar<int> lanes;
+  if (op->type.lanes() != 1) {
+    TVM_TRY_REWRITE(!broadcast(x, lanes), broadcast(!x, lanes));
+  }
+
+  TVM_TRY_REWRITE(!(!x), x);
+  TVM_TRY_REWRITE(!(x <= y), y < x);
+  TVM_TRY_REWRITE(!(x >= y), x < y);
+  TVM_TRY_REWRITE(!(x < y), y <= x);
+  TVM_TRY_REWRITE(!(x > y), x <= y);
+  TVM_TRY_REWRITE(!(x == y), x != y);
+  TVM_TRY_REWRITE(!(x != y), x == y);
```

Comment:
consider adding `TVM_TRY_REWRITE(!(x || y), (!x) && (!y));` and may be the dual for &&.